TestAggregateQueries.cls

```diff
@@ -0,0 +1,203 @@
+         }
+         // Start the Test
+         Test.startTest();
+         
+         // Insert the List
+         insert opportunityList;
+         
+         // Check the records are added or not
+         System.assertEquals(4,opportunityList.size());
+         
+         // Stop the test
+         Test.stopTest();
+         // Create the Instance of the Class
+         AggregateRelationShipSoql aggregateInstance = new AggregateRelationShipSoql();
+         Map<Integer,Double> mapInstance=aggregateInstance.sumOfClosedOpportunityAmountForCurrentFiscalYear();
+         System.assertEquals(54000,mapInstance.get(2015)); 
+    } 
+    
+    /**
+     * This is a method that test to retrieve sum of all Open Opportunity amount for 
+     * current fiscal year. Store this information in a map with key as year and value as 
+     * sum of amount. Iterate this map to display statistics.
+     */
+     static testMethod void testOpportunityRecordsUnsuccessful() 
+     {
+         // Create the instance of the Opportunity
+         List<Opportunity> opportunityList = new List<Opportunity>();
+         for (Integer i = 0; i<=1; i++) 
+         {
+             // Create the opportunity Record
+             Opportunity opportunityInstance = new Opportunity(Name = 'Test' + i,
+                                                               CloseDate = System.Today(),
+                                                               StageName = 'Qualification',
+                                                               Amount = 13000);
+                                                               
+             // Create the opportunity Record
+             Opportunity opportunityInstance2 = new Opportunity(Name = 'Test' + i,
+                                                                CloseDate = System.Today(),
+                                                                StageName = 'Qualification',
+                                                                Amount = 14000);
+             // Add records to the List
+             opportunityList.add(opportunityInstance);
+             opportunityList.add(opportunityInstance2);
+         }
+         // Start the Test
+         Test.startTest();
+         
+         // Insert the List
+         insert opportunityList;
+         
+         // Check the records are added or not
+         System.assertEquals(4,opportunityList.size());
+         
+         // Stop the test
+         Test.stopTest();
+         // Create the Instance of the Class
+         AggregateRelationShipSoql aggregateInstance = new AggregateRelationShipSoql();
+         Map<Integer,Double> mapInstance=aggregateInstance.sumOfClosedOpportunityAmountForCurrentFiscalYear();
+         System.assertEquals(null,mapInstance.get(2015)); 
+    } 
+    
+   /**
+    * This is a Method that Test to get distinct Lead records on basis of 'Lead Source'
+    *  having greater than 10 leads. Print this information.
```

Comment:
Please read and understand what you are doing, then implement.
Correct the comment